src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -168,28 +174,31 @@ class RegularSync(
     */
   def handleNewBlockHashesMessages: Receive = {
     case MessageFromPeer(NewBlockHashes(hashes), peerId) =>
-      val maybePeer = peersToDownloadFrom.find(peer => peer._1.id == peerId)
-      //we allow asking for new hashes when we are not syncing and we can download from specified peer ,we are
-      //top of the chain and not resolving branches currently
-      if (notDownloading() && topOfTheChain && maybePeer.isDefined) {
-        log.debug("Handling NewBlockHashes message: \n" + hashes.mkString("\n"))
-        val (peer, _) = maybePeer.get
-        val hashesToCheck = hashes.take(syncConfig.maxNewHashes)
-
-        if (!containsAncientBlockHash(hashesToCheck)) {
-          val filteredHashes = getValidHashes(hashesToCheck)
-
-          if (filteredHashes.nonEmpty) {
-            val request = GetBlockHeaders(Right(filteredHashes.head.hash), filteredHashes.length, BigInt(0), reverse = false)
-            requestBlockHeaders(peer, request)
-            cancelScheduledResume()
+      if (!importingBlocks) {
+        val maybePeer = peersToDownloadFrom.find(peer => peer._1.id == peerId)
+        //we allow asking for new hashes when we are not syncing and we can download from specified peer ,we are
+        //top of the chain and not resolving branches currently
+        if (notDownloading() && topOfTheChain && maybePeer.isDefined) {
+          log.debug("Handling NewBlockHashes message: \n" + hashes.mkString("\n"))
+          val (peer, _) = maybePeer.get
+          val hashesToCheck = hashes.take(syncConfig.maxNewHashes)
+
+          if (!containsAncientBlockHash(hashesToCheck)) {
+            val filteredHashes = getValidHashes(hashesToCheck)
+
+            if (filteredHashes.nonEmpty) {
+              val request = GetBlockHeaders(Right(filteredHashes.head.hash), filteredHashes.length, BigInt(0), reverse = false)
+              requestBlockHeaders(peer, request)
+              importingBlocks = true
```

Comment:
Because `newBlockHashes` message, start whole reqest block cycle (`requestHeaders->checkHeaders->requestBodies->importblocks`) and i do not want any other import block during this time becouse it would probably leave blockchain in bad state.